internal/hashing/argon2id.go

```diff
@@ -0,0 +1,162 @@
+package hashing
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"golang.org/x/crypto/argon2"
+)
+
+const ARGON2_VERSION = 19
+
+type Argon2IDParameters struct {
+	SaltLength uint32
```

Comment:
Why don't we include the version in the params?

Reply:
It was not in there because it is basically the only supported version by Go and made no sense as a parameter, but with the encoding changes, it might make sense to bring it in.

Reply:
I also removed the version constant from this file and instead used the one from the standard library.